xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -665,9 +661,13 @@ bool CGUIMediaWindow::GetDirectory(const std::string &strDirectory, CFileItemLis
 
     if (strDirectory.empty())
       SetupShares();
-
-    if (!m_rootDir.GetDirectory(pathToUrl, items))
+    
+    CFileItemList dirItems;
+    if (!m_rootDir.GetDirectory(pathToUrl, dirItems))
       return false;
```

Comment:
i had a look at the old implementation and there we don't clear the items list in case we can't fetch the directory.